sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchIndexingPublisher.java

```diff
@@ -182,48 +202,70 @@ int getBatchSize() {
                     );
                 }
 
-                return Flux.just(new IndexBatchResponse(null, actionsOffset, actions.size(), true));
+                return Flux.just(new IndexBatchResponse(statusCode, null, actionsOffset, actions.size(), true));
             });
     }
 
     private void handleResponse(List<TryTrackingIndexAction<T>> actions, IndexBatchResponse batchResponse) {
         /*
          * Batch has been split until it had one document in it and it returned a 413 response.
          */
-        if (batchResponse.getResults() == null && batchResponse.getCount() == 1) {
+        if (batchResponse.getStatusCode() == HttpURLConnection.HTTP_ENTITY_TOO_LARGE && batchResponse.getCount() == 1) {
             IndexAction<T> action = actions.get(batchResponse.getOffset()).getAction();
             onActionErrorBiConsumer.accept(action,
                 new RuntimeException("Document is too large to be indexed and won't be tried again."));
             return;
         }
 
+        List<TryTrackingIndexAction<T>> actionsToRetry = new ArrayList<>();
+        boolean has503 = batchResponse.getStatusCode() == HttpURLConnection.HTTP_UNAVAILABLE;
         if (batchResponse.getResults() == null) {
-            return;
-        }
+            /*
+             * Null results indicates that the entire request failed. Retry all documents.
+             */
+            int offset = batchResponse.getOffset();
+            actionsToRetry.addAll(actions.subList(offset, offset + batchResponse.getCount()));
```

Comment:
does the order of action matters?

Reply:
The actions here will retain the same ordering as they were submitted